Fehler beim Ausführen aus anderem Ordner

Wenn `rexlint` aus einem anderem Ordner aufgerufen wird, können teilweise Skripte nicht korrekt geladen werden (falscher Pfad).

```
$ /Users/marcel/.composer/vendor/bin/rexlint application/

PHP checks
----------

sh: /Users/marcel/Public/www/project/vendor/bin/parallel-lint: No such file or directory
sh: line 0: exec: /Users/marcel/Public/www/project/vendor/bin/parallel-lint: cannot execute: No such file or directory

 [ERROR] PHP checks failed



JSON checks
-----------

find: vendor/bin/jsonlint: No such file or directory

 [ERROR] JSON checks failed



SQL checks
----------

application/xxxx.sql
/Users/marcel/.composer/vendor/friendsofredaxo/linter/bin/lint-file.sh: line 7: vendor/bin/lint-query: No such file or directory
cat: stdout: Broken pipe

 [ERROR] SQL checks failed



'vendor/bin/yaml-lint' 'application/xxxx/.travis.yml'
sh: /Users/marcel/Public/www/project/vendor/bin/yaml-lint: No such file or directory
sh: line 0: exec: /Users/marcel/Public/www/project/vendor/bin/yaml-lint: cannot execute: No such file or directory
YAML checks
-----------


 [ERROR] YAML checks failed



```

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,12 +8,13 @@ before_install:
 
 script:
   - composer install
+  # negative tests
   - php bin/rexlint tests/fail/yml/   > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne  1 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/fail/php/   > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne  2 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/fail/json/  > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne  4 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/fail/sql/   > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne 16 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/fail/mixed/ > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne  7 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
+  
+  # positive tests
   - php bin/rexlint tests/succeed/
-
-  - cd /
-  - php $TRAVIS_BUILD_DIR/bin/rexlint $TRAVIS_BUILD_DIR/tests/succeed/
+  - cd / && php $TRAVIS_BUILD_DIR/bin/rexlint $TRAVIS_BUILD_DIR/tests/succeed/

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,3 +14,6 @@ script:
   - php bin/rexlint tests/fail/sql/   > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne 16 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/fail/mixed/ > /dev/null 2>&1 ; lintstatus=$?; if [ $lintstatus -ne  7 ]; then echo "expected fail, got $lintstatus" && exit 50; fi
   - php bin/rexlint tests/succeed/
+
+  - cd /
+  - php $TRAVIS_BUILD_DIR/bin/rexlint $TRAVIS_BUILD_DIR/tests/succeed/

--- a/bin/lint-file.sh
+++ b/bin/lint-file.sh
@@ -1,7 +1,9 @@
 #!/usr/bin/env sh
 
+BASEDIR=$(dirname $0)
+
 ## wrapper bash file which allows to pass a filepath to lint-query
 
 echo "$1"
 
-cat "$1" | vendor/bin/lint-query
+cat "$1" | $BASEDIR/../vendor/bin/lint-query

--- a/lib/Command/LintCommand.php
+++ b/lib/Command/LintCommand.php
@@ -25,6 +25,12 @@ final class LintCommand extends Command
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $rootPath = dirname(__FILE__, 3);
+
+        // If package vendor folder isn't available use project vendor folder
+        if (!file_exists($rootPath.'/vendor')) {
+            $rootPath = dirname($rootPath, 3);
+        }
+
         // some lecture on "find -exec vs. find | xargs"
         // https://www.everythingcli.org/find-exec-vs-find-xargs/
 

--- a/lib/Command/LintCommand.php
+++ b/lib/Command/LintCommand.php
@@ -27,7 +27,7 @@ final class LintCommand extends Command
         $rootPath = dirname(__FILE__, 3);
 
         // If package vendor folder isn't available use project vendor folder
-        if (!file_exists($rootPath.'/vendor')) {
+        if (!is_dir($rootPath.'/vendor')) {
             $rootPath = dirname($rootPath, 3);
         }
 

--- a/lib/Command/LintCommand.php
+++ b/lib/Command/LintCommand.php
@@ -24,6 +24,7 @@ final class LintCommand extends Command
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $rootPath = dirname(__FILE__, 3);
         // some lecture on "find -exec vs. find | xargs"
         // https://www.everythingcli.org/find-exec-vs-find-xargs/
 
@@ -33,17 +34,17 @@ final class LintCommand extends Command
         $processes[] = [
             self::ERR_PHP,
             'PHP checks',
-            $this->asyncProc(['vendor/bin/parallel-lint', '--exclude', 'vendor', $dir]),
+            $this->asyncProc([$rootPath.'/vendor/bin/parallel-lint', '--exclude', 'vendor', $dir]),
         ];
         $processes[] = [
             self::ERR_JSON,
             'JSON checks',
-            $this->asyncProc(['find', $dir, '-type', 'f', '-name', '*.json', '!', '-path', '*/vendor/*', '-exec', 'vendor/bin/jsonlint', '{}', '+']),
+            $this->asyncProc(['find', $dir, '-type', 'f', '-name', '*.json', '!', '-path', '*/vendor/*', '-exec', $rootPath.'/vendor/bin/jsonlint', '{}', '+']),
         ];
         $processes[] = [
             self::ERR_SQL,
             'SQL checks',
-            $this->asyncProc(['find', $dir, '-name', '*.sql', '!', '-path', '*/vendor/*', '-exec', dirname(__FILE__, 3) .'/bin/lint-file.sh', '{}', '+']),
+            $this->asyncProc(['find', $dir, '-name', '*.sql', '!', '-path', '*/vendor/*', '-exec',  $rootPath.'/bin/lint-file.sh', '{}', '+']),
         ];
 
         // $this->syncProc(['npm', 'install', 'csslint']);
@@ -91,7 +92,7 @@ final class LintCommand extends Command
 
         // yaml-lint only supports one file at a time
         $label = 'YAML checks';
-        $succeed = $this->syncFindExec(['find', $dir, '-type', 'f', '-name', '*.yml', '!', '-path', '*/vendor/*'], ['vendor/bin/yaml-lint']);
+        $succeed = $this->syncFindExec(['find', $dir, '-type', 'f', '-name', '*.yml', '!', '-path', '*/vendor/*'], [$rootPath.'/vendor/bin/yaml-lint']);
 
         if (!$succeed) {
             $style->section('YAML checks');

--- a/lib/Command/LintCommand.php
+++ b/lib/Command/LintCommand.php
@@ -27,6 +27,8 @@ final class LintCommand extends Command
         $rootPath = dirname(__FILE__, 3);
 
         // If package vendor folder isn't available use project vendor folder
+        // In Dev Env the vendor-binaries folder is located in the project root
+        // If this package is loaded via composer the vendor-binaries folder is located in project root and not in linter root
         if (!is_dir($rootPath.'/vendor')) {
             $rootPath = dirname($rootPath, 3);
         }

